modules/oe_theme_content_event/src/Plugin/ExtraField/Display/StatusMessage.php

```diff
@@ -0,0 +1,204 @@
+   *   Time based cache tag generator service.
+   * @param \Drupal\Core\Datetime\DateFormatterInterface $date_formatter
+   *   The date formatter.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager, TimeInterface $time, TimeBasedCacheTagGeneratorInterface $cache_tag_generator, DateFormatterInterface $date_formatter) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $entity_type_manager, $time, $cache_tag_generator);
+    $this->dateFormatter = $date_formatter;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager'),
+      $container->get('datetime.time'),
+      $container->get('oe_time_caching.time_based_cache_tag_generator'),
+      $container->get('date.formatter')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getLabel() {
+    return $this->t('Status message');
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+   * @SuppressWarnings(PHPMD.NPathComplexity)
+   */
+  public function viewElements(ContentEntityInterface $entity) {
+    // Render event status messages using the ECL Message component.
+    $event = new EventNodeWrapper($entity);
+    $build = ['#theme' => 'oe_theme_content_event_status_message'];
+
+    $cacheable = CacheableMetadata::createFromRenderArray($build);
+
+    // If the event is not over then apply time-based tags, so that it can be
+    // correctly invalidated once the event is over.
+    if (!$event->isOver($this->requestDateTime)) {
+      $this->applyHourTag($build, $event->getEndDate());
+    }
+
+    // If we have a link for livestream the livestream messages have priority
+    // over the event status ones.
+    if (!$entity->get('oe_event_online_link')->isEmpty()) {
+      // Use the 'warning' variant with the 'livestream' icon.
+      $build['#variant'] = 'warning';
+      $build['#icon'] = 'livestreaming';
+
+      // The livestream is over, but the event is still ongoing.
+      if ($event->isOnlineOver($this->requestDateTime) && $this->requestDateTime >= $event->getStartDate()->getPhpDateTime()) {
+        // Cache it by the event end date and add the timezone cache contexts.
+        $build['#title'] = $this->t('The livestream has ended, but the event is ongoing.');
+        $this->applyHourTag($build, $event->getEndDate());
+        $cacheable->addCacheContexts(['timezone']);
+        $cacheable->applyTo($build);
+        return $build;
+      }
+
+      // The event is ongoing but the livestream is yet to start.
+      if ($this->requestDateTime >= $event->getStartDate()->getPhpDateTime() && $this->requestDateTime < $event->getOnlineStartDate()->getPhpDateTime()) {
```

Comment:
We will re-use condition often, so EventNodeWrapper is a good place to store it. You can use isOnlinePeriodYetToCome from https://github.com/openeuropa/oe_content/pull/498.

`$this->requestDateTime >= $event->getStartDate()->getPhpDateTime()` also can be stored in EventNodeWrapper.